app/models/script.rb

```diff
@@ -0,0 +1,24 @@
+class Script < ApplicationRecord
+
+  has_many :events
+
+  before_validation :add_or_update_checksum
+
+  validates_presence_of :text
+  validate :unique_text
+
+  private
+
+  def add_or_update_checksum
+    return nil unless self.text # avoid triggering error on Script.new.valid?
+    checksum = Digest::SHA1.hexdigest(self.text)
+    if self.checksum.nil? || self.checksum != checksum
+      self.checksum = checksum
+    end
+  end
```

Comment:
Nothing quick, and certainly nothing that minimizes DB size. Given the constraints you outlined let's just go with hashing. Coffee, sleep and it not being 11:45PM do a lot to improve my PR feedback.